### PR TITLE
Add location filter chip in search

### PR DESCRIPTION
**Pull request by lookacat**
## Description
See https://github.com/owncloud/web/issues/9258

## Related Issue

- Fixes https://github.com/owncloud/web/issues/9258

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by lookacat** [comment]
The glitch when selecting a different location scope while the search page is open is related to the backend not yet working, it just stops the update of the component for some reason. So probably fixed as soon as the backend works.

File: packages/design-system/src/components/OcSearchBarFilter/OcSearchBarFilter.vue [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            <oc-icon name="check" />
+          </div>
+        </oc-button> </template
+    ></oc-filter-chip>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, watch } from 'vue'
+import 'vue-select/dist/vue-select.css'
```
Comment (JammingBen) [inline packages/design-system/src/components/OcSearchBarFilter/OcSearchBarFilter.vue:34]: Is this line needed?

File: packages/design-system/src/components/OcSearchBarFilter/OcSearchBarFilter.vue [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    return { currentSelection, isIndexGreaterZero, userSelected }
+  },
+  methods: {
+    onOptionSelected(option: string) {
+      this.userSelected = option
+      this.currentSelection = option
+      this.$emit('update:modelValue', { value: option })
+    }
+  }
```
Comment (JammingBen) [inline packages/design-system/src/components/OcSearchBarFilter/OcSearchBarFilter.vue:80]: Composition API please 🙂

File: packages/web-app-files/src/components/Search/List.vue [hunk 3]
```diff
@@ -276,11 +282,8 @@ export default defineComponent({
       () => unref(route).query,
       (newVal, oldVal) => {
         const filters = ['q_fullText', 'q_tags']
-        const isChange =
-          newVal?.term !== oldVal?.term ||
-          filters.some((f) => newVal[f] ?? undefined !== oldVal[f] ?? undefined)
 
-        if (isChange && isLocationCommonActive(router, 'files-common-search')) {
```
Comment (JammingBen) [inline packages/web-app-files/src/components/Search/List.vue:283]: Why did you remove all that code? It now fires a search on every route change, we should avoid that. Wouldn't it be sufficient to add a check if the `useScope` param has changed?

File: packages/web-app-search/src/portals/SearchBar.vue [hunk 4]
```diff
@@ -97,20 +99,49 @@
...
+        enabled: true,
+        default: false
+      },
+      {
+        title: $gettext('Current Folder'),
+        id: SearchLocationFilterConstants.currentFolder,
+        enabled: currentFolderAvailable,
+        default: true
+      }
+    ])
```
Comment (JammingBen) [inline packages/web-app-search/src/portals/SearchBar.vue:144]: Would be nice to have a type somewhere so we can re-use it across the app.

File: packages/web-pkg/src/composables/search/constants.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export abstract class SearchLocationFilterConstants {
+  static readonly defaultModeName: string = 'current-folder'
+  static readonly allFiles: string = 'all-files'
+  static readonly currentFolder: string = 'current-folder'
+  static readonly queryName: string = 'location-filter'
```
Comment (JammingBen) [inline packages/web-pkg/src/composables/search/constants.ts:5]: This query name is not used anywhere, it it?

**Review by JammingBen — CHANGES_REQUESTED** [review]

**Review by JammingBen — APPROVED** [review]
Nice work! 🥳

**Pull request merged**